cpp/dataStructures/CMakeLists.txt

```diff
@@ -1,16 +1,12 @@
-cmake_minimum_required(VERSION 3.2.2)
-set(CMAKE_BUILD_TYPE Release)
-
-#For the shared library:
-set ( PROJECT_LINK_LIBS libcppCommon.dylib )
-link_directories( ../common/build )
+# However, the file(GLOB...) allows for wildcard additions:
+file(GLOB SOURCES "src/*.cpp")
 
-#Bring the headers, such as Student.h into the project
-include_directories(include)
-include_directories(../common/include/)
+# Generate the shared library from the sources
+add_library(cppDS SHARED ${SOURCES})
 
```

Comment:
I notice that you add the target first and then include directories later on. Any specific reason for doing it like this or is it more of a convention ?